.github/workflows/dev-image.yml

```diff
@@ -0,0 +1,52 @@
+name: Docker Image - Development
+
+on:
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  get_semver:
+      name: Create SemVer
+      runs-on: ubuntu-latest
+      steps:
+        - name: Checkout
+          uses: actions/checkout@v2
+          with:
+            fetch-depth: 0
+        - name: Install GitVersion
+          uses: gittools/actions/gitversion/setup@v0.9.7
+          with:
+            versionSpec: "5.x"
+        - name: Determine Version
+          id: gitversion
+          uses: gittools/actions/gitversion/execute@v0.9.7
+      outputs:
+        semVer: ${{ steps.gitversion.outputs.semVer }}
+
+  build_and_publish_image:
+    name: Build Docker Image
+    runs-on: ubuntu-latest
+    needs: get_semver
+    steps:
+      - name: Checkout 
+        uses: actions/checkout@v3
+      - name: Login to GHCR
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Build and Push
+        uses: docker/build-push-action@v2
+        with:
+          context: ./
+          file: ./Dockerfile
+          builder: ${{ steps.buildx.outputs.name }}
+          push: true
+          build-args: VERSION=${{ needs.get_semver.outputs.semVer }}
+          tags: |
+            ghcr.io/jakobottar/ava:${{ needs.get_semver.outputs.semVer }}
+            ghcr.io/jakobottar/ava:dev
+            ghcr.io/jakobottar/ava:latest
```

Comment:
You are missing a third job here which is to create a github tag after you publish your docker image. your image is being tagged with like (v0.1.12) but that tag is not associated with any commit in source code. If you needed to create a hotfix you have no idea what commit to branch from. Creating a tag in Github will better tie docker tags to your code.

Reply:
That makes sense yeah I would keep it simple. Do you really even need to build PRs then? I normally will setup a build action for main then a pr-checks action for PRs that just runs tests.